Consul/Consul.csproj

```diff
@@ -45,23 +25,21 @@
     <DefineConstants>$(DefineConstants);CORECLR</DefineConstants>
   </PropertyGroup>
 
-  <Target Name="ILMerge" AfterTargets="AfterBuild" Condition=" '$(TargetFramework)' == 'net461' ">
+  <Target Name="ILRepack" AfterTargets="AfterBuild" Condition=" '$(TargetFramework)' == 'net461' ">
     <ItemGroup>
-      <BuildArtifacts Include="$(OutputPath)$(AssemblyName).*" />
-      <ArtifactsToMerge Include="$(OutputPath)Newtonsoft.Json.dll" />
+      <ArtifactsToMerge Include="Newtonsoft.Json.dll" />
     </ItemGroup>
     <PropertyGroup>
-      <IlmergeCommand>$(ILMergeConsolePath) /internalize /out:$(OutputPath)ILMerge\$(AssemblyName).dll</IlmergeCommand>
-      <IlmergeCommand>$(IlmergeCommand) /keyfile:$(AssemblyOriginatorKeyFile)</IlmergeCommand>
-      <IlmergeCommand>$(IlmergeCommand) $(OutputPath)$(AssemblyName).dll @(ArtifactsToMerge->'%(RelativeDir)%(filename)%(extension)', ' ')</IlmergeCommand>
+      <ILRepackCommand>$(ILRepack) /parallel /internalize /out:ILRepack/$(AssemblyName).dll</ILRepackCommand>
+      <ILRepackCommand>$(ILRepackCommand) /keyfile:$([System.IO.Path]::GetFullPath('$(AssemblyOriginatorKeyFile)'))</ILRepackCommand>
+      <ILRepackCommand>$(ILRepackCommand) $(AssemblyName).dll @(ArtifactsToMerge->'%(filename)%(extension)', ' ')</ILRepackCommand>
+      <ILRepackCommand Condition=" '$(Channel)' == 'mono' ">mono $(ILRepackCommand)</ILRepackCommand>
     </PropertyGroup>
-    <MakeDir Directories="$(OutputPath)\ILMerge" />
-    <Exec Command="$(IlmergeCommand)" ContinueOnError="false" />
-    <Move SourceFiles="%(BuildArtifacts.RelativeDir)ILMerge\%(filename)%(extension)"
-          DestinationFiles="%(BuildArtifacts.RelativeDir)%(filename)%(extension)"
-          OverwriteReadOnlyFiles="true"/>
-    <Delete Files="@(ArtifactsToMerge)" />
-    <RemoveDir Directories="$(OutputPath)ILMerge" />
+    <MakeDir Directories="$(OutputPath)/ILRepack" />
+    <Exec WorkingDirectory="$(OutputPath)" Command="$(ILRepackCommand)" ContinueOnError="false" StandardOutputImportance="normal" />
+    <Move SourceFiles="$(OutputPath)ILRepack/$(AssemblyName).dll" DestinationFiles="$(OutputPath)$(AssemblyName).dll" OverwriteReadOnlyFiles="true" />
+    <RemoveDir Directories="$(OutputPath)ILRepack" />
+    <Message Importance="high" Text="Merged @(ArtifactsToMerge->'%(filename)%(extension)', ', ') into $(OutputPath)$(AssemblyName).dll" />
```

Comment:
Can we extract this change into separate PR ? I I assume that it is being done because `ILMerge` repository is now archived ?

Reply:
Not sure, likely related to Mono https://github.com/jgiannuzzi/consuldotnet/commit/559e332d4374f3403de4f5fea6fc49a298e93426